lib/Server.js

```diff
@@ -9,24 +9,40 @@
 const http = require('http');
 const EventEmitter = require('events');
 
-const DataStore = require('./stores/DataStore');
 const GetHandler = require('./handlers/GetHandler');
 const HeadHandler = require('./handlers/HeadHandler');
 const OptionsHandler = require('./handlers/OptionsHandler');
 const PatchHandler = require('./handlers/PatchHandler');
 const PostHandler = require('./handlers/PostHandler');
 const DeleteHandler = require('./handlers/DeleteHandler');
 const RequestValidator = require('./validators/RequestValidator');
+const Uid = require('./models/Uid');
 const EXPOSED_HEADERS = require('./constants').EXPOSED_HEADERS;
 const REQUEST_METHODS = require('./constants').REQUEST_METHODS;
 const TUS_RESUMABLE = require('./constants').TUS_RESUMABLE;
 const debug = require('debug');
 const log = debug('tus-node-server');
 class TusServer extends EventEmitter {
 
-    constructor() {
+    constructor(options) {
         super();
 
+        if (!options) {
+            throw new Error('\'options\' must be defined');
+        }
+        if (!options.path) {
+            throw new Error('\'path\' is not defined must have a path');
+        }
+        if (options.namingFunction && typeof options.namingFunction !== 'function') {
+            throw new Error('\'namingFunction\' must be a function');
+        }
+
+        this.options = { ...options };
+        if (!this.options.namingFunction) {
+            this.options.namingFunction = Uid.rand;
+        }
```

Comment:
Move this to PostHandler as it is only used there. Without `creation` extension it is not even needed.